sip/sip-006-runtime-cost-assessment.md

```diff
@@ -746,3 +754,130 @@ types is defined as:
 * `map`: the size of stored key + the size of the stored value.
 * `nft`: the size of the NFT key
 * `ft`: the size of a Clarity uint value.
+
+# Cost Upgrades
+
+In order to enable the addition of new native functions to the Clarity
+language, there is a mechanism for voting on and changing a function's
+_cost-assessment function_ (the asymptotic function that describes its
+complexity and is used to calculate its cost). 
+
+New functions can be introduced to Clarity by first being implemented
+_in_ Clarity and published in a contract. This is necessary to avoid
+a hard fork of the blockchain and ensure the availability of the function
+across the network.
+
+If it is decided that a published function should be a part of the Clarity
+language, it can then be re-implemented as a native function in Rust and
+included in a future release of Clarity. Nodes running this upgraded VM
+would instead use this new native implementation of the function when they
+encounter Clarity code that references it via `contract-call?`.
+
+The new native function is likely faster and more efficient than the
+prior Clarity implementation, so a new cost-assessment function may need
+to be chosen for its evaluation. Until a new cost-assessment function is
+agreed upon, the cost of the Clarity implementation will continue to be used.
+
+## Voting on the Cost of Clarity Functions
+
+New and more accurate cost-assessment functions can be agreed upon as
+follows:
+
+1. A user formulates a cost-assessment function and publish it in a
+Clarity contract as a `read-only` function.
+2. The user proposes the cost-assessment function by calling the
+`propose-cost` function in the **Clarity Cost Voting Contract**.
+3. Voting on the proposed function ensues via the voting functions in
+the **Clarity Cost Voting Contract**, and the function is either
+selected as a replacement for the existing cost-assessment function,
+or ignored.
+
+### Publishing a Cost-Assessment Function
+
+Cost-assessment functions to be included in proposals can be published
+in Clarity contracts. They must be written precisely as follows:
+
+1. The function should be `read-only`.
+2. The function should return a tuple with the keys `runtime`, `write_length`,
+`write_count`, `read_count`, and `read_length` (the execution cost measurement
+[categories](#measurements-for-execution-cost)).
+3. The values of the returned tuple should be Clarity expressions representing
+the asymptotic cost functions.
+3. The expression's constants and variables should be listed as arguments
+to the Clarity function.
+    * The arguments should be written in alphabetical order, and variables
+    should come before constants.
+
+Here is an example of a cost-assessment function where the runtime
+is set to `a * log(n) + b`, and all of the other parameters are constant.
+
+```Lisp
+(define-read-only (runtime_cost (n int) (a int) (b int))
```

Comment:
I'm assuming that the reason the function creator creates an associated runtime cost function is because the Clarity VM will eventually use it to determine how much block capacity the Rust version of the function will use?  If so, then what values are in the `n`, `a`, and `b` variables here?  Do cost functions like this need to have a given signature?